frontend/src/components/UserGroupCard/index.tsx

```diff
@@ -1,53 +1,88 @@
 import React, { useEffect, useState } from "react";
 import "./styles.css";
-import { fetchUsersWithRole } from "@/api/users";
+import { fetchUser, fetchUsersWithRole } from "@/api/users";
 
 interface IUserGroupCardProps {
   label: string;
   role_type: string;
   classroom: IClassroom;
+  givenUsersList?: IClassroomUser[];
   onClick?: () => void;
 }
 
 const UserGroupCard: React.FC<IUserGroupCardProps> = ({
   label,
   role_type,
   classroom,
+  givenUsersList,
```

Comment:
I wonder if you could set this to a default [] instead of having null checks later on

Reply:
the point of it being nullable is that it's an optional field. You could make the argument that it should be mandatory, but making it default empty doesn't make sense to me.

Reply:
If no users are provided, then you have an empty list of given users. It just simplifies your code later on so you're not manually handling null cases (i.e. length = 0 when emty, does exist, etc.)